src/bluesearch/entrypoint/database/parse.py

```diff
@@ -107,22 +131,45 @@ def run(
     input_type: str,
     input_path: Path,
     output_dir: Path,
+    match_filename: Optional[str],
+    recursive: bool,
+    dry_run: bool,
 ) -> int:
     """Parse one or several articles.
 
     Parameter description and potential defaults are documented inside of the
     `get_parser` function.
     """
     inputs: Iterable[Path]
+
     if input_path.is_file():
         inputs = [input_path]
+
     elif input_path.is_dir():
-        inputs = sorted(input_path.glob("*"))
+        if recursive:
+            pattern = "**/*"
+        else:
+            pattern = "*"
+        files = (x for x in input_path.glob(pattern) if x.is_file())
+
+        if match_filename:
+            regex = re.compile(match_filename)
+            selected = (x for x in files if regex.fullmatch(x.name))
+        else:
+            selected = files
```

Comment:
Maybe we can get rid of this else-block by re-assigning `files` in the if-block?
```python
files = (x for x in files if regex.fullmatch(x.name))
```

Reply:
Re-assigning a variable is not a good practice from my experience.

Anyway, the if-else block has now a new condition. So I guess that is also more readable to keep the block now.